### ISSUE TITLE
CwlPreconditionTesting dependency doesn't work

### ISSUE THREAD

**Issue by danielsaidi** [Quick/Nimble#696]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/master/CONTRIBUTING.md) and have done my best to follow them.

## TL;DR

Nimble `8.0.4` depends on `CwlPreconditionTesting`, which makes builds break. I understand why you want to move out code from Nimble and that base it on a library like `CwlPreconditionTesting`, but I find it to be a dealbreaker if it means that it will make Nimble less stable and force all libraries that depend on even more external dependencies.

### What did you do?

My libraries (uses SPM) updated their dependencies to the latest Nimble version, `8.0.4`. This made the builds break.

Forcing SPM to use `8.0.2` makes everything work again.

### What did you expect to happen?

Libraries that upgrade to `8.0.4` from `8.0.2` should build and run like before.

### What actually happened instead?

My libraries fail to build, since `CwlPreconditionTesting` fail with `Undefined symbols for architecture x86_64`.

### Environment

List the software versions you're using:

 - Quick: 2.2.0
 - Nimble: 8.0.4
 - Xcode Version: Xcode 11 GM 2
 - Swift Version: 5.1

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Cocoapods: -
 - Carthage: -
 - Swift Package Manager: Apple Swift Package Manager - Swift 5.1.0 (swiftpm-14696)

**Comment by ikesyo** [comment]
Thanks for raising the issue. Actually Nimble has been using `CwlPreconditionTesting` dependency except for the SwiftPM integration until 8.0.3/8.0.4. We added the dependency for SwiftPM to support the `throwAssertion` matcher in SwiftPM (see https://github.com/Quick/Nimble/issues/674 for the details). Unfortunately it now appears that Xcode 11's SwiftPM support is a bit different from SwiftPM usages on CLI projects so far. I'll take care of this in a next major version. Until then, please use 8.0.2 since there is no so much differences between the versions.

**Comment by danielsaidi** [comment]
Ah ok, I see...yeah it would be great if you could find a way to support `throwAssertion` as well. I'll go back to `8.0.2` and hope that you manage to get it working. Good luck!

**Comment by inetdavid** [comment]
I was able to get commit fc3cd1adf6961a3541679342d64fbbfd7a9f4ca7 working using Xcode 11 by loading the package using the commit hash, which is two commits after tag v8.0.4.  Did the latest version fail to get tagged?

**Comment by danielsaidi** [comment]
@ikesyo Do you have a release plan för 8.0.5 and do you plan on including the fix in that version?

**Comment by ikesyo** [comment]
The latest CwlPreconditionTesting doesn't support Swift 4.2 so it's a breaking change for us, thus we need major version bump.

**Comment by danielsaidi** [comment]
Ok, thanks for letting me know!

**Comment by danielsaidi** [comment]
@ikesyo This is still an issue. I'm not sure why it's closed? Upgrading to `Nimble 8.0.5` in my SPM projects makes the test run fail with the following error:

```
Showing All Messages
xctest (27033) encountered an error (Failed to load the test bundle. (Underlying error: The bundle “SheeeeeeeeetTests” couldn’t be loaded because it is damaged or missing necessary resources. The bundle is damaged or missing necessary resources. dlopen_preflight(/Users/admin/Library/Developer/Xcode/DerivedData/Sheeeeeeeeet-bpdwemkusvsybpefrvxivlepzywu/Build/Products/Debug-iphonesimulator/SheeeeeeeeetTests.xctest/SheeeeeeeeetTests): Library not loaded: @rpath/CwlPreconditionTesting.framework/CwlPreconditionTesting
  Referenced from: /Users/admin/Library/Developer/Xcode/DerivedData/Sheeeeeeeeet-bpdwemkusvsybpefrvxivlepzywu/Build/Products/Debug-iphonesimulator/SheeeeeeeeetTests.xctest/SheeeeeeeeetTests
  Reason: image not found))
```

Reverting back to `8.0.2` makes the tests run again. I run the latest Xcode and Swift versions.

**Comment by ikesyo** [comment]
The issue was closed when merging #706. Could you please use `master` branch until v9 is released.

**Comment by danielsaidi** [comment]
I can test and see if it works.
